validation/src/main/scala/hmda/validation/dsl/PredicateGeo.scala

```diff
@@ -0,0 +1,87 @@
+package hmda.validation.dsl
+
+import hmda.model.census.{ CBSAMetroMicroLookup, CBSATractLookup }
+import hmda.model.fi.lar.Geography
+
+object PredicateGeo {
+  private val cbsaTracts = CBSATractLookup.values
+
+  private val cbsaMetroMicro = CBSAMetroMicroLookup.values
+
+  private val validMsaCombinationSet = cbsaTracts.map { cbsa =>
+    (cbsa.geoIdMsa, cbsa.state, cbsa.county, cbsa.tractDecimal)
+  }.toSet
+
+  private val validMdCombinationSet = cbsaTracts.map { cbsa =>
+    (cbsa.metDivFp, cbsa.state, cbsa.county, cbsa.tractDecimal)
+  }.toSet
+
+  private val validMsaCombinationSetNoTract = cbsaTracts.map { cbsa =>
+    (cbsa.geoIdMsa, cbsa.state, cbsa.county)
+  }.toSet
+
+  private val validMdCombinationSetNoTract = cbsaTracts.map { cbsa =>
+    (cbsa.metDivFp, cbsa.state, cbsa.county)
+  }.toSet
+
+  private val MsaNotMicro = cbsaMetroMicro
+    .filter(_.MEMI == 1)
+    .map { _.GEOIOD }
```

Comment:
This probably should be a Set, though I haven't measured anything.  (And for that matter, I don't know whether anything here is the most efficient algorithm, and it'd be premature to do much optimizing.  Still, I suspect a Set would be good, and certainly the code would look simpler if everything is a Set. Otherwise it invites the question, why not this one?)

Also, names.  I was about to suggest making this one start with a lower-case letter, but perhaps these are actually all constants and it'd be better to change all the others!  Hm.